src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -35,7 +35,7 @@
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public abstract void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
```

Comment:
https://github.com/jenkinsci/jenkins/commit/cb6e568815976a2309bff9ebf3e4f7e0a070e769#diff-f53a4bf03f87b35d03b5c29476b723edR42 is the one case where I can remember actually testing this properly.